### PR TITLE
Move to the official .NET Core 2.0

### PR DESCRIPTION
**Pull request by daxian-dbw**
.NET Core 2.0 was [officially released](https://blogs.msdn.microsoft.com/dotnet/2017/08/14/announcing-net-core-2-0/) on 8/14/2017.
This PR is to migrate our tooling and build to the official .NET Core 2.0 bits.

### CODE REVIEW

File: appveyor.yml [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 image: Visual Studio 2017
 
-# cache version - netcoreapp.2.0.0-preview3-25426-01
+# cache version - netcoreapp.2.0.0
 cache:
   - '%LocalAppData%\Microsoft\dotnet -> appveyor.yml'
```
Comment (TravisEz13) [inline appveyor.yml]: Why don't we update the caches to depend on `PowerShell.Common.props`.  Then we only have one place to update the version.

**Review by TravisEz13 — APPROVED** [review]
1 Minor comment.

File: appveyor.yml (hunk 1 above)
Comment (daxian-dbw) [inline appveyor.yml]: There are some changes pending for `PowerShell.Common.props` (see Ilya's MSBuild refactoring PRs). I think for the time being, `PowerShell.Common.props` has more churns than `appveyor.yml`.

**Review by adityapatwardhan — APPROVED** [review]
LGTM

**Comment by adityapatwardhan** [comment]
@daxian-dbw This is a major change, should we run Feature tests?

**Comment by daxian-dbw** [comment]
@adityapatwardhan I requested feature test run for the PR, please take a look at the second commit message. I will investigate failure in appveyor.

**Comment by daxian-dbw** [comment]
There were 11 tests failing in AppVeyor and all are web cmdlet tests. They failed because of time out.

**Review by mirichmo — APPROVED** [review]

File: appveyor.yml (hunk 1 above)
Comment (TravisEz13) [inline appveyor.yml]: sounds good

**Pull request merged**